src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -2141,56 +2143,47 @@ def client_side_proxy(cmd,
-        else:
-            custom_locations_oid = get_custom_locations_oid(cmd, cl_oid)
-            if custom_locations_oid == "":
-                enable_custom_locations = False
     except Exception as e:
-        enable_custom_locations = False
-        logger.warning("Unable to fetch registration state of 'Microsoft.ExtendedLocation'. Failed to enable 'custom-locations' feature. This is fine if not required. Proceeding with helm install.")
+        logger.warning("Unable to fetch registration state of 'Microsoft.ExtendedLocation'. Failed to enable 'custom-locations' feature.")
         telemetry.set_exception(exception=e, fault_type=consts.Custom_Locations_Registration_Check_Fault_Type,
                                 summary='Unable to fetch status of Custom Locations RP registration.')
-    return enable_custom_locations, custom_locations_oid
+        raise
+    if cl_registration_state != "Registered":
+        logger.warning("Custom locations couldn't be enabled on this cluster as 'Microsoft.ExtendedLocation' is not registered. More details for enabling this feature can be found here: https://aka.ms/EnableCustomLocations")
+        return False
+    return True
 
 
-def get_custom_locations_oid(cmd, cl_oid):
+def custom_locations_oid_valid(cmd, cl_oid):
+    if cl_oid is None:
+        raise RequiredArgumentMissingError("The passed in custom locations object id is None. Please pass in the custom locations object id to verify.")
     try:
         sp_graph_client = get_graph_client_service_principals(cmd.cli_ctx)
-        sub_filters = []
-        sub_filters.append("displayName eq '{}'".format("Custom Locations RP"))
-        result = list(sp_graph_client.list(filter=(' and '.join(sub_filters))))
-        if len(result) != 0:
-            if cl_oid is not None and cl_oid != result[0].object_id:
-                logger.debug("The 'Custom-locations' OID passed is different from the actual OID({}) of the Custom Locations RP app. Proceeding with the correct one...".format(result[0].object_id))
-            return result[0].object_id  # Using the fetched OID
-
-        if cl_oid is None:
-            logger.warning("Failed to enable Custom Locations feature on the cluster. Unable to fetch Object ID of Azure AD application used by Azure Arc service. Try enabling the feature by passing the --custom-locations-oid parameter directly. Learn more at https://aka.ms/CustomLocationsObjectID")
-            telemetry.set_exception(exception='Unable to fetch oid of custom locations app.', fault_type=consts.Custom_Locations_OID_Fetch_Fault_Type,
-                                    summary='Unable to fetch oid for custom locations app.')
-            return ""
-        else:
-            return cl_oid
+        # Do not use display names for look-up as display names are not unique.
+        result = list(sp_graph_client.list(filter=("objectId eq '{}'".format(cl_oid))))
```

Comment:
minor nit, but oid is the key to get a service principal so can query directly for object rather than doing a list query with filter. also, what is the high-level flow here, as it is necessary to get a graph token from user to call graph to lookup this SP just to trace the display name? perhaps this should be skipped entirely?

Reply:
It uses the Azure CLI context to call the AAD graph to get the SP app information.